konfetti/src/main/java/nl/dionsegijn/konfetti/Confetti.kt

```diff
@@ -96,26 +95,17 @@ class Confetti(
             return
         }
 
-        var left = location.x + (width - rotationWidth)
-        var right = location.x + rotationWidth
-        /** Switch values. Left or right may not be negative due to how Android Api < 25
-         *  draws Rect and RectF, negative values won't be drawn resulting in flickering confetti */
-        if (left > right) {
-            left += right
-            right = left - right
-            left -= right
-        }
-
         paint.alpha = alpha
 
-        rectF.set(left, location.y, right, location.y + getSize())
```

Comment:
Moved all of this logic into canvas operations so shape implementations don't need to worry about it and can just draw normally.

Reply:
Moving the drawing operations makes sense, thanks!